### PR TITLE
Unity: Snap Schedule Support

### PR DESCRIPTION
**Pull request by Murray-LIANG**

### CODE REVIEW

File: storops/unity/resource/lun.py [hunk 1]
```diff
@@ -515,6 +529,9 @@ def replicate_with_dst_resource_provisioning(self, max_time_out_of_sync,
             self._cli, self.get_id(), dst_resource, max_time_out_of_sync,
             remote_system=remote_system, name=replication_name)
 
+    def remove_snap_schedule(self):
+        return self.modify(is_snap_schedule_paused=False)
```
Comment (anil-degwekar) [inline storops/unity/resource/lun.py]: Should this be True?

Comment (Murray-LIANG) [reply storops/unity/resource/lun.py]: `False` is correct. I am confusing about this too. But I dumped the REST request sent by Unisphere. `False` is used there too.

I am consulting with the feature team about the detail. Will update this comment once I have the answer.

File: storops_test/unity/resource/test_snap_schedule.py [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+         'is_auto_delete': True, 'retention_time': 3600,
+         'access_type': FilesystemSnapAccessTypeEnum.CHECKPOINT,
+         'expected':
+             {'type': ScheduleTypeEnum.N_HOURS_AT_MM, 'minute': 30,
+              'hours': None, 'daysOfWeek': None,
+              'daysOfMonth': None,
+              'interval': 3, 'isAutoDelete': True, 'retentionTime': 3600,
+              'accessType': FilesystemSnapAccessTypeEnum.CHECKPOINT}},
+    )
+    def test_every_n_hours(self, kwargs):
```
Comment (yong-huang) [inline storops_test/unity/resource/test_snap_schedule.py]: No negative test case for every_n_hours.

File: storops_test/unity/resource/test_snap_schedule.py [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+         'is_auto_delete': True, 'retention_time': 3600,
+         'access_type': FilesystemSnapAccessTypeEnum.CHECKPOINT,
+         'expected':
+             {'type': ScheduleTypeEnum.N_DAYS_AT_HHMM, 'minute': 30,
+              'hours': [0], 'daysOfWeek': None,
+              'daysOfMonth': None,
+              'interval': 3, 'isAutoDelete': True, 'retentionTime': 3600,
+              'accessType': FilesystemSnapAccessTypeEnum.CHECKPOINT}},
+    )
+    def test_every_n_days(self, kwargs):
```
Comment (yong-huang) [inline storops_test/unity/resource/test_snap_schedule.py]: No negative test case for every_n_days.

File: storops_test/unity/resource/test_snap_schedule.py [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+        {'days_of_month': 10, 'hour': None, 'minute': None,
+         'is_auto_delete': None, 'retention_time': None, 'access_type': None,
+         'error_msg': 'days_of_month of UnitySnapScheduleRule every_month '
+                      'rule should be a list.'},
+        {'days_of_month': [], 'hour': None, 'minute': None,
+         'is_auto_delete': None, 'retention_time': None, 'access_type': None,
+         'error_msg': 'days_of_month of UnitySnapScheduleRule every_month '
+                      'rule cannot be empty list.'}
+    )
+    def test_every_month_exception(self, kwargs):
```
Comment (yong-huang) [inline storops_test/unity/resource/test_snap_schedule.py]: Why not follow the naming rule of the previous negative cases: test_xxx_value_error

File: storops_test/unity/resource/test_snap_schedule.py (hunk 2 above)
Comment (Murray-LIANG) [reply storops_test/unity/resource/test_snap_schedule.py]: `every_n_hours` and `every_n_days` are just invoking `to_embedded` directly without extra logic. So, testing `to_embedded` with negative cases is enough.

File: storops_test/unity/resource/test_snap_schedule.py (hunk 3 above)
Comment (Murray-LIANG) [reply storops_test/unity/resource/test_snap_schedule.py]: ditto.

File: storops_test/unity/resource/test_snap_schedule.py (hunk 4 above)
Comment (Murray-LIANG) [reply storops_test/unity/resource/test_snap_schedule.py]: Changed it to `test_every_month_value_error`.

**Pull request merged**